src/sif/pedestriancost.cc

```diff
@@ -633,11 +633,16 @@ bool PedestrianCost::AllowedReverse(const baldr::DirectedEdge* edge,
 Cost PedestrianCost::EdgeCost(const baldr::DirectedEdge* edge,
                               const baldr::GraphTile* tile,
                               const uint32_t seconds) const {
+  auto speed = tile->GetSpeed(edge, flow_mask_, seconds);
 
+  if (shortest_) {
+    float sec = edge->length() * (kSecPerHour * 0.001f) / static_cast<float>(speed);
+    return Cost(edge->length(), sec);
+  }
+  
   // Ferries are a special case - they use the ferry speed (stored on the edge)
   if (edge->use() == Use::kFerry) {
-    auto speed = tile->GetSpeed(edge, flow_mask_, seconds);
-    float sec = edge->length() * (kSecPerHour * 0.001f) / static_cast<float>(speed);
+    float sec = edge->length() * (kSecPerHour * 0.001f) / static_cast<float>(edge->speed());
```

Comment:
seemed wrong here

Reply:
i dont think its wrong actually. the get speed function will eventually return edge->speed, it just makes sure to check if there isnt any higher resolution speed information on the edge. i think it should probably stay as it was in case in the future we get better information about ferries and their speeds at different times of day (unlikely but would remove a gotcha)

Reply:
gotcha, reviewed the other ferry costs as well and reverted what I did there